src/ImageSharp.Drawing/Processing/Drawing/Processors/FillProcessor.cs

```diff
@@ -49,38 +50,61 @@ protected override void OnFrameApply(ImageFrame<TPixel> source, Rectangle source
             int minY = Math.Max(0, startY);
             int maxY = Math.Min(source.Height, endY);
 
-            // Reset offset if necessary.
-            if (minX > 0)
-            {
-                startX = 0;
-            }
-
-            if (minY > 0)
-            {
-                startY = 0;
-            }
-
             int width = maxX - minX;
 
-            using (IBuffer<float> amount = source.MemoryManager.Allocate<float>(width))
-            using (BrushApplicator<TPixel> applicator = this.brush.CreateApplicator(
-                source,
-                sourceRectangle,
-                this.options))
-            {
-                amount.Span.Fill(this.options.BlendPercentage);
+            var solidBrush = this.brush as SolidBrush<TPixel>;
 
+            // If there's no reason for blending, then avoid it.
+            if (solidBrush != null &&
+                (
```

Comment:
I refactor this logic into an `IsSolidBrushWithoutBlending` property or similar for better readability.